GVFS/GVFS.Platform.Windows/WindowsPlatform.cs

```diff
@@ -266,6 +268,29 @@ public override bool TryInstallGitCommandHooks(GVFSContext context, string execu
             return true;
         }
 
+        public override bool TryVerifyAuthenticodeSignature(PhysicalFileSystem fileSystem, string path, out string subject, out string error)
+        {
+            using (Stream stream = fileSystem.OpenFileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, false))
+            using (PowerShell powershell = PowerShell.Create())
+            {
+                powershell.AddScript($"Get-AuthenticodeSignature -FilePath {path}");
+
+                Collection<PSObject> results = powershell.Invoke();

```

Comment:
Will take a look at this

Reply:
Couldn't find an option in this API to specify `-NonInteractive -NoProfile` parameters. Wrote the following script to check whether PowerShell is launched interactive and also to find the profile that is loaded. Pasting the code used to test - 

```
public static void CheckInteractiveAndUserProfile()
        {
            RunCommand1($"Read-Host Please enter your name:");
            Console.WriteLine("");
            RunCommand1($"$profile");
        }

        public static void RunCommand1(string command)
        {
            using (PowerShell powershell = PowerShell.Create())
            {
                powershell.AddScript(command);
                Collection<PSObject> results = powershell.Invoke();
                if (powershell.HadErrors || results.Count <= 0)
                {
                    foreach (ErrorRecord error in powershell.Streams.Error)
                    {
                        Console.WriteLine($"{command} Error: {error.Exception.Message}");
                    }
                }
                else
                {
                    foreach (PSObject result in results)
                    {
                        string output = result == null ? "null" : result.ToString();
                        Console.WriteLine($"{command}: {output}");
                    }
                }
            }
        }
```

This is the output:
```
Read-Host Please enter your name: Error: A command that prompts the user failed because the host program or the command type does not support user interaction. The host was attempting to request confirmation with the following message:

$profile: null
```

While using `ProcessHelper.Run("powershell.exe")` method, for the same `Read-Host` and `$profile` commands I get similar results -
```
Read-Host Please enter your name: Error: Read-Host : Windows PowerShell is in NonInteractive mode. Read and Prompt functionality is not available.
At line:1 char:5
+ & { Read-Host Please enter your name: }
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Read-Host], PSInvalidOperationException
    + FullyQualifiedErrorId : InvalidOperation,Microsoft.PowerShell.Commands.ReadHostCommand



$profile: Output: C:\Users\alsha\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1
```

Summary

| Launch Option | powershell.exe | PowerShell API |
| ------------- | ------------- |------------- |
| NonInteractive | No user interaction allowed | No user interaction allowed |
| NoProfile | Loads $HOME\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1 | Loads No profile |

Reply:
> Loads $HOME\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1

How did you confirm powershell loaded the profile?  The documentation says: 

> -NoProfile
> Doesn't load the PowerShell profile.
https://docs.microsoft.com/en-us/powershell/scripting/components/console/powershell.exe-command-line-help?view=powershell-6

>THE NOPROFILE PARAMETER
>To start PowerShell without profiles, use the NoProfile parameter of PowerShell.exe, the program that starts PowerShell.
>https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-6

Is is possible the `$profile` variable is set even though the profile was not loaded?

Could you try adding something to your profile that would make it obvious it's being added?

Reply:
> Could you try adding something to your profile that would make it obvious it's being added?

Note: The main concern I had was with the `PowerShell` in `System.Management.Automation`, there's no need to further investigate the behavior of powershell.exe for this PR.